lib/msf/core/exploit/remote/kerberos/service_authenticator/options.rb

```diff
@@ -21,7 +21,7 @@ def kerberos_auth_options(protocol:, auth_methods:)
         'Kerberos encryption types to offer',
         Msf::Exploit::Remote::AuthOption::KERBEROS_DEFAULT_OFFERED_ENC_TYPES.join(',')
       ],
-      regex: Regexp.new("(#{etype_regex},)*#{etype_regex}", options: Regexp::IGNORECASE),
+      regex: Regexp.new("(#{etype_regex},)*#{etype_regex}", Regexp::IGNORECASE),
```

Comment:
Context:

Ruby 3.2:

```
3.2.0 :005 > Regexp.new("foo", options: Regexp::IGNORECASE)
(irb):5:in `initialize': unknown keyword: :options (ArgumentError)
	from (irb):5:in `new'
	from (irb):5:in `<main>'
	from /Users/adfoster/.rvm/rubies/ruby-3.2.0/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
	from /Users/adfoster/.rvm/rubies/ruby-3.2.0/bin/irb:25:in `load'
	from /Users/adfoster/.rvm/rubies/ruby-3.2.0/bin/irb:25:in `<main>'
```

Ruby 3.1
```
>> Regexp.new("foo", options: Regexp::IGNORECASE)
=> /foo/i
```

Both versions of ruby work with just options passed in

```
3.2.0 :006 > Regexp.new("foo", Regexp::IGNORECASE)
 => /foo/i
```